src/main/java/solutions/siren/join/common/Bytes.java

```diff
@@ -114,4 +126,34 @@ public final static long readLong(BytesRef src) {
     return (((long) readInt(src)) << 32) | (readInt(src) & 0xFFFFFFFFL);
   }
 
+  public final static void writeBytesRef(BytesRef src, BytesRef dst) {
+    if (src == null) {
```

Comment:
Why in this condition we write a zero to dst ? It seems like an unexpected side-effect to calling `writeBytesRef`. If it is OK, maybe document it in the method doc ?